### PR TITLE
raftstore: count kvs in other cf when doing split check

### PR DESCRIPTION
**Pull request by BusyJay**
We only count the kvs in default cf when doing split check, which can lead to a region with too many keys if most values are very small. So we need to count all the kvs in other large cfs too.

### CODE REVIEW

File: src/raftstore/store/worker/split_check.rs [hunk 1]
```diff
@@ -80,17 +170,23 @@ impl Runnable<Task> for Runner {
...
+        let res = MergedIterator::new(task.engine.as_ref(),
+                                      LARGE_CFS,
+                                      &task.start_key,
+                                      &task.end_key,
+                                      false)
+            .map(|mut iter| {
+                while let Some(e) = iter.next() {
+                    size += e.len() as u64;
+                    if split_key.is_empty() && size > self.split_size {
+                        split_key = e.key.unwrap();
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/split_check.rs]: Is `split_key` user key? Or contains ts tail?

Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/split_check.rs]: ignore

File: src/raftstore/store/worker/split_check.rs [hunk 2]
```diff
@@ -13,18 +13,108 @@
...
+            iters.push(iter);
+        }
+        Ok(MergedIterator {
+            iters: iters,
+            heap: heap,
+        })
+    }
+
+    #[allow(should_implement_trait)]
+    fn next(&mut self) -> Option<KeyEntry> {
```
Comment (siddontang) [inline src/raftstore/store/worker/split_check.rs]: why not traveling the vector iterators one by one? maybe we don't need heap here.

File: tests/raftstore/test_split_region.rs [hunk 3]
```diff
@@ -138,10 +145,10 @@ fn test_auto_split_region<T: Simulator>(cluster: &mut Cluster<T>) {
 
     assert_eq!(region, target);
 
-    let final_key = put_till_size(cluster,
-                                  REGION_MAX_SIZE - REGION_SPLIT_SIZE + check_size_diff,
-                                  &mut range);
-    let max_key = cmp::max(last_key, final_key);
+    let max_key = put_cf_till_size(cluster,
```
Comment (siddontang) [inline tests/raftstore/test_split_region.rs]: do we have test for `default` cf now?

File: src/raftstore/store/worker/split_check.rs (hunk 2 above)
Comment (BusyJay) [inline src/raftstore/store/worker/split_check.rs]: Then how do you maintain the order of output keys?

File: tests/raftstore/test_split_region.rs (hunk 3 above)
Comment (BusyJay) [inline tests/raftstore/test_split_region.rs]: Yes.

File: src/raftstore/store/worker/split_check.rs [hunk 4]
```diff
@@ -80,17 +170,23 @@ impl Runnable<Task> for Runner {
...
-            }
-            Ok(size < self.region_max_size)
-        });
+        let res = MergedIterator::new(task.engine.as_ref(),
+                                      LARGE_CFS,
+                                      &task.start_key,
+                                      &task.end_key,
+                                      false)
+            .map(|mut iter| {
+                while let Some(e) = iter.next() {
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/split_check.rs]: The first key/value pair not count.

**Comment by zhangjinpeng1987** [comment]
rest LGTM

File: src/raftstore/store/worker/split_check.rs (hunk 4 above)
Comment (BusyJay) [inline src/raftstore/store/worker/split_check.rs]: Why？

File: src/raftstore/store/worker/split_check.rs (hunk 2 above)
Comment (siddontang) [inline src/raftstore/store/worker/split_check.rs]: do we need to maintain the order? The output keys for same cf is in order, I think it is ok.

File: src/raftstore/store/worker/split_check.rs (hunk 4 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/split_check.rs]: my mistake, ignore

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**